eppo_core/src/events/event_dispatcher.rs

```diff
@@ -26,6 +27,12 @@ pub struct EventDispatcherConfig {
     pub max_retries: Option<u32>,
 }
 
+#[derive(Debug)]
+pub enum DispatcherError {
+    QueueError(QueueError),
+    EventDeliveryError(&'static str),
```

Comment:
minor: please prefer named errors + display implementation (see `thiserror`) instead of generic str slices. Strings are hard for the caller to analyze programmatically to decide what to do with the error